src/System.Memory/tests/Span/CovariantSliceNotSupported.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.MemoryTests;
+using Xunit;
+
+namespace System.SpanTests
+{
+    public static partial class SpanTests
+    {
+        [Fact]
+        public static void CovariantSlicesNotSupported1()
```

Comment:
We already have such ctor tests for span (CovariantSlicesNotSupported1 / CovariantSlicesNotSupported3): https://github.com/dotnet/corefx/blob/ec1671fd7feffe1070a94d9d86379e0dd95ddcdf/src/System.Memory/tests/Span/CtorArray.cs#L97-L98

The only one that needs to be added is CovariantSlicesNotSupported2, which is testing AsSpan. Therefore, we should move that one with the rest of the AsSpan tests. 
CovariantAsSpanNotSupported


```C#
        [Fact]
        public static void CovariantAsSpanNotSupported()
        {
            object[] array = new string[10];
            Assert.Throws<ArrayTypeMismatchException>(() => array.AsSpan());
            Assert.Throws<ArrayTypeMismatchException>(() => array.AsSpan(0, a.Length));
        }
```